lib/services/sqs.js

```diff
@@ -118,8 +118,27 @@ AWS.util.update(AWS.SQS.prototype, {
    * @api private
    */
   buildEndpoint: function buildEndpoint(request) {
-    var params = request.httpRequest.params || this.config.params;
-    var url = params.QueueUrl;
+    var self = this;
+
+    var params = (function () {
+      // Query protocol client or request level parameters
+      if (request.httpRequest.params && request.httpRequest.params.QueueUrl) {
+        return request.httpRequest.params;
+      }
+      // JSON protocol with config QueueUrl.
+      if (self.config.params && self.config.params.QueueUrl) {
+        return self.config.params;
```

Comment:
But if users from the Query version put QueueUrl into params, it does not get serialized into the body, it only exists on this config.params object. 

In contrast, a client param in Query is serialized into its outgoing requests.

Reply:
resolved in revision: added query compat customization to JSON protocol